fxa-oauth-server/bin/purge_expired_tokens.js

```diff
@@ -58,38 +61,48 @@ const ignorePocketClientId = program.pocketId.toLowerCase().split(/\s*,\s*/g);
 
 db.ping().done(() => {
   // Only mysql impl supports token deletion at the moment
-  if (! db.purgeExpiredTokens) {
+  if (!(db.purgeExpiredTokens && db.purgeExpiredTokensById)) {
     const message = ('Unable to purge expired tokens, only available ' +
                      'when using config with mysql database.');
     logger.info('skipping', { message: message });
     return;
   }
 
-  logger.info('deleting', {
-    numberOfTokens: numberOfTokens,
-    delaySeconds: delaySeconds,
-    deleteBatchSize: deleteBatchSize,
-    ignorePocketClientId: ignorePocketClientId
-  });
-
-  // To reduce the risk of deleting pocket tokens, ensure that the pocket-id
-  // passed in belongs to a client.
+  // To reduce the risk of deleting pocket tokens, purgeExpiredTokens(ById?)
+  // will ensure that the pocket-id passed in belongs to a client.
   const purgeMethod = program.byId ? db.purgeExpiredTokensById : db.purgeExpiredTokens;
-  return purgeMethod(numberOfTokens,
-                     delaySeconds,
-                     ignorePocketClientId,
-                     deleteBatchSize)
-    .then(() => {
-      logger.info('completed');
-      process.exit(0);
-    })
-    .catch((err) => {
-      logger.error('error', err);
+
+  return db.getLock(DB_GET_LOCK_STRING, DB_GET_LOCK_TIMEOUT)
+    .then((result) => {
+      logger.info('getLock', { result: result });
+
+      if (result.acquired !== 1) {
+        logger.info('getLock', { error: `Could not acquire cooperative lock '${DB_GET_LOCK_STRING}'` });
+        process.exit(1);
+      }
+
+      logger.info('deleting', {
+        numberOfTokens: numberOfTokens,
+        delaySeconds: delaySeconds,
+        deleteBatchSize: deleteBatchSize,
+        ignorePocketClientId: ignorePocketClientId
+      });
+      return purgeMethod(numberOfTokens,
+                         delaySeconds,
+                         ignorePocketClientId,
+                         deleteBatchSize)
+        .then(() => {
+          logger.info('completed');
+          process.exit(0);
+        })
+        .catch((err) => {
+          logger.error('error', err);
+          process.exit(1);
+        });
+    }, (err) => {
+      logger.critical('db.ping', err);
       process.exit(1);
     });
-}, (err) => {
-  logger.critical('db.ping', err);
-  process.exit(1);
```

Comment:
It feels like we should have two `(err) =>`  blocks here, one for failure in `db.ping` and one for failure in `db.getLock`?

Reply:
Yep